cmd/akash/deployment.go

```diff
@@ -98,23 +112,25 @@ func createDeployment(ctx context.Context, cmd *cobra.Command, args []string) er
 		return errors.New(res.DeliverTx.GetLog())
 	}
 
-	fmt.Println(X(deployment.Address))
+	address := res.DeliverTx.Data
+
+	fmt.Println(X(address))
 
 	if ctx.Wait() {
 		fmt.Printf("Waiting...\n")
-		expected := len(groups.GetItems())
+		expected := len(groups)
 		handler := marketplace.NewBuilder().
 			OnTxCreateFulfillment(func(tx *types.TxCreateFulfillment) {
-				if bytes.Equal(tx.Fulfillment.Deployment, deployment.Address) {
+				if bytes.Equal(tx.Fulfillment.Deployment, address) {
 					f := tx.Fulfillment
-					fmt.Printf("Group %v/%v Fulfillment: %X\n", f.Group, len(groups.GetItems()),
+					fmt.Printf("Group %v/%v Fulfillment: %X\n", f.Group, len(groups),
 						state.FulfillmentID(f.Deployment, f.Group, f.Order, f.Provider))
 				}
 			}).
 			OnTxCreateLease(func(tx *types.TxCreateLease) {
-				if bytes.Equal(tx.Lease.Deployment, deployment.Address) {
+				if bytes.Equal(tx.Lease.Deployment, address) {
 					l := tx.Lease
-					fmt.Printf("Group %v/%v Lease: %X\n", l.Group, len(groups.GetItems()),
+					fmt.Printf("Group %v/%v Lease: %X\n", l.Group, len(groups),
```

Comment:
hmm not sure how how did `%X` got back in there.  use `"%v",X(state.FulfillmentID(...)))` for these.